mlflow/projects/__init__.py

```diff
@@ -66,13 +67,18 @@ def _run(uri, entry_point="main", version=None, parameters=None, experiment_id=N
     # Add branch name tag if a branch is specified through -version
     if _is_valid_branch_name(work_dir, version):
         tracking.MlflowClient().set_tag(active_run.info.run_uuid, MLFLOW_GIT_BRANCH_NAME, version)
+        git_remote_url = _get_remote_repo_url(work_dir)
```

Comment:
Why not have this top level instead of just if it's a valid branch? Does it not work in that case? People might have a repo URL but not be on a branch.

Reply:
@mateiz Sorry , but moving it to toplevel causes `test_run_databricks` to fail. Would it make sense to move it  inside `elif mode == "local" or mode is None:` as the `databricks` mode can stay separate and since this is an OR clause tags could be recorded for all mode != `databricks`

Reply:
I can fix it up to work in all modes if desired. This shouldn't work any differently in the two modes -- it sounds like you just need to change that test because it assumes a certain set of tags will be present in a certain order, but we now added a new tag.